src/helpers/chartHelpers.tsx

```diff
@@ -3,38 +3,43 @@ import { isNumber } from "lodash";
 import { ChartThreshold, MultiSeriesLineChartOptions } from "./chartOptions";
 import { resolveColor } from "./colors";
 import { LayoutContext } from "../components";
-
-export function createLineChartDefs(layoutContext: LayoutContext, gradientKey: string, seriesLineColors: string[], multiSeriesLineChartOptions?: MultiSeriesLineChartOptions) {
-    let nDomainMin: number | undefined = undefined;
-    let nDomainMax: number | undefined = undefined;
-
-    if (multiSeriesLineChartOptions) {
-        if (isNumber(multiSeriesLineChartOptions.domainMin)) nDomainMin = multiSeriesLineChartOptions.domainMin;
-        nDomainMax = multiSeriesLineChartOptions.domainMax;
-    }
-
-    const getPercent = function (numerator: number, domainMin: number, domainMax: number): number {
-        if ((domainMax - domainMin) > 0){
-            return (numerator - domainMin) / (domainMax - domainMin) * 100;
+import { TimeSeriesDataPoint } from '../components/presentational/TimeSeriesChart/TimeSeriesChart';
+
+export function createLineChartDefs(
+    layoutContext: LayoutContext, 
+    gradientKey: string, 
+    seriesLineColors: string[], 
+    multiSeriesLineChartOptions: MultiSeriesLineChartOptions | undefined,
+    dataKeys: string[],
+    data: TimeSeriesDataPoint[] ) {
+
+    const getPercent = function (numerator: number, min: number, max: number) {
+        if ((max - min) > 0){
+            return (numerator - min) / (max - min) * 100;
         }
         return 0;
     }
 
-    const createStopsFromThresholds = function (defaultLineColor: string, domainMin: number, domainMax: number, chartThresholds?: ChartThreshold[]) {
+    const createStopsFromThresholds = function (defaultLineColor: string, chartThresholds: ChartThreshold[] | undefined, index: number) {
         let stops: any[] = [];
         var lineColor: string = defaultLineColor;
         var thresholds = chartThresholds ?? [];
 
         thresholds.sort((a, b) => a.value - b.value);
 
-        if (thresholds.length && domainMin >= thresholds[0].value) {
+        const lineRange = data.reduce((result, dataPoint) => ({
```

Comment:
The domain max was initially passed in so that the chart could continue letting the y max be auto calculated, while still being able to know the range for the sake of calculating gradient stops. 

For single lines where the y domain is passed in, can the re calculation of max/min be avoided? In the case of [ hr intraday  ](https://github.com/CareEvolution/MyDataHelpsUI/blob/029e0e4cb7b9a7f6ee6839e7ce14e62733dfb5ab/src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx#L95) , it already doing the leg work of figuring out what the max should be.

Reply:
I think that I would argue that in a follow up PR to this one, that we could remove the calculation from the HR graph completely, and let it set the domain max to auto.  But, that somewhat depends on what the exact behavior from the HR graph needs to be.  It's something we can have further conversation on, and would potentially impact the change made here https://github.com/CareEvolution/MyDataHelpsUI/pull/281/files#diff-c7d85721f79b358b3ab7238e978537aa98ffcf42fc45afbb6a1eb497b473892fR225 to be a configurable behavior.

However, from the perspective of the TimeSeriesChart as a reusable widget, I think it's correct for this calculation to be down here in the utility functions; I wouldn't want to push the responsibility of calculating the line min/max to parent components because it's a detail that's very tied to exactly how we're rendering the threshold colors.